plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -557,7 +557,7 @@ void saveProgram() throws IOException {
     @CpsVmThreadOnly
     public void saveProgram(File f) throws IOException {
         File dir = f.getParentFile();
-        File tmpFile = File.createTempFile("atomic",null, dir);
+        File tmpFile = Files.createTempFile(dir.toPath(), "atomic", null).toFile();
```

Comment:
This is not equivalent and will create a file with more restrictive permissions than desired.
As the file is moved later on (line 580) it will retain this restrictive permissions.
Thus I do not believe this to be a correct change.

Reply:
Also, what problem does this PR purport to solve, other than simply applying some automated refactoring rule? The title implies something about “security” but if there is a vulnerability, it should be reported through proper channels, never as a public PR.